admin/api/routes.go

```diff
@@ -34,37 +34,68 @@ func (h *RoutesHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	hostFilter := getFilterParameter(r, "host")
+	pathFilter := getFilterParameter(r, "path")
+	tagFilter := getFilterParameter(r, "tag")
+
 	var hosts []string
-	for host := range t {
-		hosts = append(hosts, host)
+
+	if hostFilter != nil {
+		hosts = append(hosts, *hostFilter)
+	} else {
+		for host := range t {
+			hosts = append(hosts, host)
+		}
+		sort.Strings(hosts)
 	}
-	sort.Strings(hosts)
 
 	var routes []apiRoute
+
 	for _, host := range hosts {
 		for _, tr := range t[host] {
-			for _, tg := range tr.Targets {
-				var opts []string
-				for k, v := range tg.Opts {
-					opts = append(opts, k+"="+v)
-				}
+			if pathFilter == nil || tr.Path == *pathFilter {
+				for _, tg := range tr.Targets {
+					if tagFilter == nil || tagsMatchFilter(tg.Tags, *tagFilter) {
```

Comment:
I think you can just encapsulate `tagFilter == nil` into the `tagsMatchFilter` helper method

Reply:
Yeah, I've been debating about this since it kind of changes the meaning of `tagsMatchFilter` to also return true if there  is no filter. I'll update.